tests/test_logfire.py

```diff
@@ -2583,3 +2539,109 @@ def test_force_flush(exporter: TestExporter):
     assert len(exporter.exported_spans_as_dict()) == 1
+
+
+def test_tags(exporter: TestExporter) -> None:
+    lf = logfire.with_tags('tag1', 'tag2')
+    with lf.span('a span', _tags=('tag2', 'tag3')):
+        lf.info('a log', _tags=('tag4', 'tag1'))
+
+    assert exporter.exported_spans_as_dict() == snapshot(
+        [
+            {
+                'name': 'a log',
+                'context': {'trace_id': 1, 'span_id': 3, 'is_remote': False},
+                'parent': {'trace_id': 1, 'span_id': 1, 'is_remote': False},
+                'start_time': 2000000000,
+                'end_time': 2000000000,
+                'attributes': {
+                    'logfire.span_type': 'log',
+                    'logfire.level_num': 9,
+                    'logfire.msg_template': 'a log',
+                    'logfire.msg': 'a log',
+                    'code.filepath': 'test_logfire.py',
+                    'code.function': 'test_tags',
+                    'code.lineno': 123,
+                    'logfire.tags': ('tag1', 'tag2', 'tag4'),
+                },
+            },
+            {
+                'name': 'a span',
+                'context': {'trace_id': 1, 'span_id': 1, 'is_remote': False},
+                'parent': None,
+                'start_time': 1000000000,
+                'end_time': 3000000000,
+                'attributes': {
+                    'code.filepath': 'test_logfire.py',
+                    'code.function': 'test_tags',
+                    'code.lineno': 123,
+                    'logfire.msg_template': 'a span',
+                    'logfire.msg': 'a span',
+                    'logfire.tags': ('tag1', 'tag2', 'tag3'),
+                    'logfire.span_type': 'span',
+                },
+            },
+        ]
+    )
+
+
+def test_tags_setter_basic_operations():
+    with logfire.span('foo') as span:
+        span.tags = ('a', 'b')
+        assert span.tags == ('a', 'b')
+
+        # Only unique tags are kept
+        span.tags += ('a',)
+        assert span.tags == ('a', 'b')
+
+        # Adding new tags
+        span.tags += ('c',)
+        assert span.tags == ('a', 'b', 'c')
+
+        span.tags += ('d',)
+        assert span.tags == ('a', 'b', 'c', 'd')
+
+
+def test_tags_setter_empty_tags_returns_empty_tuple():
+    with logfire.span('foo') as span:
+        assert span.tags == ()
+
+
+def test_tags_setter_with_non_initialized_span() -> None:
+    span = logfire.with_tags('tag1', 'tag2').span('test span')
+    assert span.tags == ('tag1', 'tag2')
+
+    # Set tags using the setter with self._span not yet created
+    span.tags = ('tag3', 'tag4')
+    assert span.tags == ('tag3', 'tag4')
+
+    # Call __enter__ to initialize _span, check that tags remain unchanged
+    with span:
+        assert span.tags == ('tag3', 'tag4')
+
+    # Check that tags remain unchanged
+    assert span.tags == ('tag3', 'tag4')
+
+
+def test_tags_setter_valid_type_extensions() -> None:
+    span = logfire.span('test span')
```

Comment:
```suggestion
    span = logfire.span('test span')
    assert span.tags == ()
```

Reply:
I added tests checking default values in many cases.